src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj

```diff
@@ -120,6 +120,7 @@
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Owin.Security" Version="4.2.0" />
```

Comment:
Is this 'legacy' OWIN dependency required?

Reply:
By any chance, did you mean the `IDataProtector` from [Microsoft.AspNetCore.DataProtection](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.idataprotector?view=aspnetcore-5.0) namespace?

Reply:
Net462 and 472 should theoretically still have the Asp.Net Core IDataProtector [available](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.Abstractions/2.0.0).

Reply:
@rymeskar: It's only for .NET 462 and .NET 472 (ASP.NET classic). If you are on .NET Core it's not added (see the conditional reference)
See src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs:

```CSharp
#if DOTNET_472 || DOTNET_462
using IDataProtectionProvider = Microsoft.Owin.Security.DataProtection.IDataProtectionProvider;
using IDataProtector = Microsoft.Owin.Security.DataProtection.IDataProtector;
#else
using IDataProtectionProvider = Microsoft.AspNetCore.DataProtection.IDataProtectionProvider;
using IDataProtector = Microsoft.AspNetCore.DataProtection.IDataProtector;
#endif
```

Reply:
My thinking here was that on Net462 and Net472, consumers can still depend on both classic ASP.NET but also on ASP.NET Core (up to version 2)

By depending on the lightweight (no dependencies) [Microsoft.AspNetCore.DataProtection.Abstractions](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.Abstractions/2.2.0) package, we are actually more versatile for the consumers. 

As for ASP.NET Core, everything is straightforward through the app platform. And for ASP.NET, consumers have two options: 
* implement the [IDataProtector](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.idataprotector?view=aspnetcore-2.0) interface with their API of choice on the given platform.
* depend on [Microsoft.AspNetCore.DataProtection](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection/2.0.0). And thus intentionally bring in the dependencies from that package.

Reply:
@jmprieur provided a PR which uses your suggested. great to delete so much code.